pkg/resourcemapping/operator_resource_mapping.go

```diff
@@ -321,102 +351,77 @@ func (ormClient *ORMClient) LocateOwnerPaths(ownedObj *unstructured.Unstructured
-		if err != nil {
-			return fmt.Errorf("failed to update CR %s for %s in namespace %s: %v", operatorRes, componentKey, resourceNamespace, err)
-		}
-		if !needsUpdate {
-			glog.V(4).Infof("No changes in controller %s/%s src path '%s'. Skip updating Operator CR %s.",
-				updatedControllerObj.GetKind(), updatedControllerObj.GetName(), srcPath, operatorRes)
-			continue
-		}
-		re := regexp.MustCompile(`(\[)|(\]\.)`)
-		parsedDestPath := re.ReplaceAllString(destPath, ".")
-		fields := strings.Split(parsedDestPath, ".")
-		if len(fields) < 2 {
-			return fmt.Errorf("failed to update %v to CR %s for %s in namespace %s: '%s' is invalid path", newValue, operatorRes, componentKey, resourceNamespace, destPath)
-		}
-		origCRValue, _, err := util.NestedField(operatorCR, resourceMappingDestPath, destPath)
-		if err != nil {
-			return fmt.Errorf("failed to update %v to CR %s '%s' for %s in namespace %s: %v", newValue, operatorRes, destPath, componentKey, resourceNamespace, err)
-		}
-		err = util.SetNestedField(operatorCR.Object, newValue, fields[1:]...)
-		if err != nil {
-			return fmt.Errorf("failed to update %v to CR %s '%s' for %s in namespace %s: %v", newValue, operatorRes, destPath, componentKey, resourceNamespace, err)
-		}
-		_, err = dynResourceClient.Update(context.TODO(), operatorCR, metav1.UpdateOptions{})
-		if err != nil {
-			return fmt.Errorf("failed to update %v to CR %s '%s' for %s in namespace %s: %v", newValue, operatorRes, destPath, componentKey, resourceNamespace, err)
+	sourceResKind := updatedControllerObj.GetKind()
+	sourceResName := updatedControllerObj.GetName()
+	sourceRes := sourceResKind + "/" + sourceResName
+	for ownedPath, resourcePaths := range ownerResources.OwnerResourcesMap {
+		for _, resourcePath := range resourcePaths {
+			// Retrieve the owner object and path
+			ownerPath := resourcePath.Path
+			ownerObj := resourcePath.ObjectReference
+			ownerResKind := ownerObj.Kind
+			ownerResName := ownerObj.Name
+			ownerRes := ownerResKind + "/" + ownerResName
+			ownerResNamespace := ownerObj.Namespace
+			// ownerResources might have source/owned resource kind with their resource paths if it cannot find the owner resource mapping from ORM.
+			// In that case we cannot perform this update operation on source/owned resource kind without owner resource found
+			if ownerResKind != updatedControllerObj.GetKind() {
+				glog.Infof("Update owner %s/%s resources found for source %s/%s",
+					ownerResKind, ownerResName,
+					sourceResKind, sourceResName)
+				ownerCR, err := kubernetes.Toolbox.GetResourceWithObjectReference(ownerObj)
+				if err != nil {
+					return fmt.Errorf("failed to get orm v2 owner CR for owner object %s in namespace %s: %v", ownerRes, ownerResNamespace, err)
+				}
+				// get the new resource value from the source obj
+				newCRValue, found, err := ormutils.NestedField(updatedControllerObj, resourceMappingSrcPath, ownedPath)
```

Comment:
This is incorrect. resourceMappingSrcPath is a pointer into the v1 ORM to get the owned resource path not in the controller Object.
The resource value you want to get from the 'updatedControllerObj' is at the ownedPath location

Reply:
Based on our previous discussion, it was decided not to introduce new constants for the 'ownedPath' or 'ownerPath' similar to 'srcPath' or 'destPath'. Instead, we have kept the path as it is to align with the ORM V1 flow. During testing, it was found that the path string we pass to the function works correctly with the JSONPath library to fetch the required resources.

This util function has changed, I will incorporate the updates mentioned in the pull request (PR), which can be found at this link: [GitHub PR link](https://github.com/turbonomic/orm/pull/36). In the new version of the function, it no longer accepts a "name" parameter; only the "obj" (object) and "path" arguments are required.